schematic/schemas/data_model_graph.py

```diff
@@ -780,16 +779,26 @@ def get_node_validation_rules(
             node_label: Label of the node for which you need to look up.
             node_display_name: Display name of the node which you want to get the label for.
         Returns:
-            A set of validation rules associated with node, as a list.
+            A set of validation rules associated with node, as a list or a dictionary.
         """
         if not node_label:
-            assert node_display_name is not None
+            if node_display_name is None:
+                raise ValueError(
+                    "Either node_label or node_display_name must be provided."
+                )
+
+            # try search node label using display name
             node_label = self.get_node_label(node_display_name)
 
         if not node_label:
             return []
 
-        node_validation_rules = self.graph.nodes[node_label]["validationRules"]
+        try:
+            node_validation_rules = self.graph.nodes[node_label]["validationRules"]
+        except:
```

Comment:
Is there a smaller more targeted exception that can be caught here rather than a catch all?

Reply:
Yeah. I just added `KeyError`